### PR TITLE
[DIG-009] Add Exponentiation

### PR DESCRIPTION
**Pull request by blondiebytes**
Why:
- Create more robust calculator by adding exponentiation feature

How:
- Modify welcome message and run-operation functions in calculator.py file to add exponentiation feature

Test Plan:
- Ensure you can use exponentiation with the calculator
- Ensure all automated tests pass

### CODE REVIEW

File: Calculator.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 def gather_inputs():
-    operation = input("Please type in operation you would like to complete: +, -, *, or /")
+    operation = input("Please type in operation you would like to complete: +, -, *, /, or ^")
```
Comment (stephanie-droid) [inline Calculator.py]: Can we state the new operation first? EX: `"^, +, -, *, or /"`

File: CalculatorTest.py [hunk 2]
```diff
@@ -24,5 +24,9 @@ def test_run_operation_invalid(self):
         with self.assertRaises(Exception):
             Calculator.run_operation(9,0, '8')
 
+    def test_run_operation_exponentiation(self):
```
Comment (stephanie-droid) [inline CalculatorTest.py:27]: Can we add a few tests that use decimal numbers and negative numbers?

File: Calculator.py (hunk 1 above)
Comment (blondiebytes) [reply Calculator.py]: Done

File: CalculatorTest.py (hunk 2 above)
Comment (blondiebytes) [reply CalculatorTest.py:27]: I'm not sure if we need additional test cases as the current tests cover how our code uses exponentiation. I believe adding additional test cases would test the built-in functionality of exponentiation rather than how our application uses it. However, if we think they are still necessary, I can add them.

File: Calculator.py (hunk 1 above)
Comment (viveksingh98) [inline Calculator.py]: Can we have an exponentiation operator as this is the desired capability for this feature request?

File: Calculator.py [hunk 3]
```diff
@@ -17,6 +17,9 @@ def run_operation(number_1,  number_2, operation):
     elif operation == '/':
         print('{} / {} = '.format(number_1, number_2), end = '')
         return number_1 / number_2
+    elif operation == '^':
+        print('{} ^ {} = '.format(number_1, number_2), end = '')
+        return number_1 ** number_2
```
Comment (viveksingh98) [inline Calculator.py:22]: Just a suggestion, can we have a switch operator instead of if else statements, as adding cases would be more readable than keeping adding if else statements?

File: CalculatorTest.py [hunk 4]
```diff
@@ -24,5 +24,9 @@ def test_run_operation_invalid(self):
...
+        self.assertEqual(16, Calculator.run_operation(2, 4, '^'))
+        self.assertEqual(1, Calculator.run_operation(5, 0, '^'))
```
Comment (viveksingh98) [inline CalculatorTest.py:29]: Can we have more scenarios like negative integers, real numbers and the like?

**Review by viveksingh98 — CHANGES_REQUESTED** [review]
Please check the changes i've mentioned and let me know if you find them useful.

File: Calculator.py (hunk 3 above)
Comment (subhadip00) [reply Calculator.py:22]: > Just a suggestion, can we have a switch operator instead of if else statements, as adding cases would be more readable than keeping adding if else statements?

Python doesn't have switch case statement.

**Review by subhadip00 — APPROVED** [review]

**Comment by pgadri** [comment]
Add ^ and ** to be both recognized as exponential so if anyone uses either of them they will get the same results

**Comment by Praveerpratap2803** [comment]
Demo

**Comment by btzodikov** [comment]
Looking for someone to help me review this code to Add Exponentiation to the python calculator. 
` elif operation == '^':
        print('{} ** {} = '.format(number_1, number_2), end = '')
        return number_1 ** number_2`

File: CalculatorTest.py (hunk 2 above)
Comment (tutumtetwa) [reply CalculatorTest.py:27]: Great!
